`report-release-versions` goal ends with NPE as no mavenArtifactResolver is set

`report-release-versions` goal ends with NPE as no mavenArtifactResolver is set

```
Caused by: java.lang.NullPointerException
    at java.util.Objects.requireNonNull (Objects.java:221)
    at io.quarkus.bom.decomposer.UpdateAvailabilityTransformer.<init> (UpdateAvailabilityTransformer.java:22)
    at io.quarkus.bom.decomposer.BomDecomposer$BomDecomposerConfig.checkForUpdates (BomDecomposer.java:95)
    at io.quarkus.bom.decomposer.maven.ReleaseVersionsReportMojo.decompose (ReleaseVersionsReportMojo.java:64)
    at io.quarkus.bom.decomposer.maven.ReleaseVersionsReportMojo.execute (ReleaseVersionsReportMojo.java:51)
```

Workaround is to set `<skipUpdates>true</skipUpdates>` in configuration of the plugin.

Naive fix:
```diff
diff --git a/maven-plugin/src/main/java/io/quarkus/bom/decomposer/maven/ReleaseVersionsReportMojo.java b/maven-plugin/src/main/java/io/quarkus/bom/decomposer/maven/ReleaseVersionsReportMojo.java
index 934465e..617f6fa 100644
--- a/maven-plugin/src/main/java/io/quarkus/bom/decomposer/maven/ReleaseVersionsReportMojo.java
+++ b/maven-plugin/src/main/java/io/quarkus/bom/decomposer/maven/ReleaseVersionsReportMojo.java
@@ -6,6 +6,8 @@ import io.quarkus.bom.decomposer.DecomposedBom;
 import io.quarkus.bom.decomposer.DecomposedBomHtmlReportGenerator;
 import io.quarkus.bom.decomposer.DecomposedBomHtmlReportGenerator.HtmlWriterBuilder;
 import io.quarkus.bom.decomposer.DecomposedBomReleasesLogger;
+import io.quarkus.bom.resolver.ArtifactResolverProvider;
+import io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver;
 import org.apache.maven.plugin.AbstractMojo;
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.MojoFailureException;
@@ -58,6 +60,12 @@ public class ReleaseVersionsReportMojo extends AbstractMojo {
         final MojoMessageWriter msgWriter = new MojoMessageWriter(getLog());
         final BomDecomposerConfig config = BomDecomposer.config()
                 .logger(msgWriter)
+                .mavenArtifactResolver(
+                        ArtifactResolverProvider
+                                .get(MavenArtifactResolver.builder()
+                                        .setOffline(true)
+                                        .setWorkspaceDiscovery(false)
+                                        .build()))
                 .debug()
                 .bomArtifact(project.getGroupId(), project.getArtifactId(), project.getVersion());
         if (!skipUpdates) {
```

--- a/bom-decomposer/src/main/java/io/quarkus/bom/resolver/DefaultArtifactResolver.java
+++ b/bom-decomposer/src/main/java/io/quarkus/bom/resolver/DefaultArtifactResolver.java
@@ -3,6 +3,7 @@ package io.quarkus.bom.resolver;
 import io.quarkus.bootstrap.model.AppArtifactCoords;
 import io.quarkus.bootstrap.resolver.maven.BootstrapMavenException;
 import io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver;
+import io.quarkus.bootstrap.resolver.maven.workspace.LocalWorkspace;
 import java.io.BufferedReader;
 import java.io.BufferedWriter;
 import java.io.File;
@@ -118,11 +119,22 @@ public class DefaultArtifactResolver implements ArtifactResolver {
             throw new ArtifactNotFoundException("Failed to describe " + a, e);
         }
         // if it didn't throw an exception it still might not exist in the local repo
-        final File file = a.getFile() == null ? new File(resolver.getSession().getLocalRepository().getBasedir(),
-                resolver.getSession().getLocalRepositoryManager().getPathForLocalArtifact(a)) : a.getFile();
-        if (!file.exists()) {
+        final boolean pomExists;
+        if (a.getFile() == null) {
+            if (new File(resolver.getSession().getLocalRepository().getBasedir(),
+                    resolver.getSession().getLocalRepositoryManager().getPathForLocalArtifact(a)).exists()) {
+                pomExists = true;
+            } else {
+                final LocalWorkspace workspace = resolver.getMavenContext().getWorkspace();
+                pomExists = workspace != null && workspace.getProject(a.getGroupId(), a.getArtifactId()) != null;
+            }
+        } else {
+            pomExists = a.getFile().exists();
+        }
+        if (!pomExists) {
             persistNotFoundArtifacts(coords);
-            throw new ArtifactNotFoundException(a + " was not found in the local repository at " + file);
+            throw new ArtifactNotFoundException(
+                    a + " was found in neither the Maven local repository nor in the project's workspace");
         }
         return result;
     }

--- a/maven-plugin/src/main/java/io/quarkus/bom/decomposer/maven/ReleaseVersionsReportMojo.java
+++ b/maven-plugin/src/main/java/io/quarkus/bom/decomposer/maven/ReleaseVersionsReportMojo.java
@@ -6,14 +6,22 @@ import io.quarkus.bom.decomposer.DecomposedBom;
 import io.quarkus.bom.decomposer.DecomposedBomHtmlReportGenerator;
 import io.quarkus.bom.decomposer.DecomposedBomHtmlReportGenerator.HtmlWriterBuilder;
 import io.quarkus.bom.decomposer.DecomposedBomReleasesLogger;
+import io.quarkus.bom.resolver.ArtifactResolverProvider;
+import io.quarkus.bootstrap.resolver.maven.MavenArtifactResolver;
+import java.util.List;
 import org.apache.maven.plugin.AbstractMojo;
 import org.apache.maven.plugin.MojoExecutionException;
 import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Component;
 import org.apache.maven.plugins.annotations.LifecyclePhase;
 import org.apache.maven.plugins.annotations.Mojo;
 import org.apache.maven.plugins.annotations.Parameter;
 import org.apache.maven.plugins.annotations.ResolutionScope;
 import org.apache.maven.project.MavenProject;
+import org.eclipse.aether.RepositorySystem;
+import org.eclipse.aether.RepositorySystemSession;
+import org.eclipse.aether.impl.RemoteRepositoryManager;
+import org.eclipse.aether.repository.RemoteRepository;
 
 @Mojo(name = "report-release-versions", defaultPhase = LifecyclePhase.VERIFY, requiresDependencyCollection = ResolutionScope.COMPILE_PLUS_RUNTIME)
 public class ReleaseVersionsReportMojo extends AbstractMojo {
@@ -42,6 +50,18 @@ public class ReleaseVersionsReportMojo extends AbstractMojo {
     @Parameter(defaultValue = "${skipBomReport}")
     protected boolean skip;
 
+    @Component
+    private RepositorySystem repoSystem;
+
+    @Component
+    private RemoteRepositoryManager remoteRepoManager;
+
+    @Parameter(defaultValue = "${project.remoteProjectRepositories}", readonly = true, required = true)
+    private List<RemoteRepository> repos;
+
+    @Parameter(defaultValue = "${repositorySystemSession}", readonly = true)
+    private RepositorySystemSession repoSession;
+
     @Override
     public void execute() throws MojoExecutionException, MojoFailureException {
         if (skip) {
@@ -57,6 +77,15 @@ public class ReleaseVersionsReportMojo extends AbstractMojo {
     private void decompose() throws Exception {
         final MojoMessageWriter msgWriter = new MojoMessageWriter(getLog());
         final BomDecomposerConfig config = BomDecomposer.config()
+                .mavenArtifactResolver(ArtifactResolverProvider.get(
+                        MavenArtifactResolver.builder()
+                                .setRepositorySystem(repoSystem)
+                                .setRepositorySystemSession(repoSession)
+                                .setRemoteRepositories(repos)
+                                .setRemoteRepositoryManager(remoteRepoManager)
+                                .setPreferPomsFromWorkspace(true)
+                                .setCurrentProject(project.getFile() == null ? null : project.getFile().getAbsolutePath())
+                                .build()))
                 .logger(msgWriter)
                 .debug()
                 .bomArtifact(project.getGroupId(), project.getArtifactId(), project.getVersion());

